### PR TITLE
Improve parallel join chapter in the book.

### PR DESCRIPTION
**Pull request by WaDelma**
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/356)

### CODE REVIEW

File: book/src/08_parallel_join.md [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 As mentioned in the chapter dedicated on how to [dispatch][c3] systems,
 Specs automatically parallelizes system execution when there are non-conflicting
-system data requirements (every system data may only be accessed
-mutably once xor immutably from multiple systems).
+system data requirements (each resource in the system data may only be accessed either
```
Comment (torkleyy) [inline book/src/08_parallel_join.md]: Maybe the text in brackets should be: "two `System`s conflict if their `SystemData` needs access to the same resource where at least one of both needs write access"

File: book/src/08_parallel_join.md [hunk 2]
```diff
@@ -45,8 +47,10 @@ fn run(&mut self, (vel, mut pos): Self::SystemData) {
 }
 ```
 
-The `par_join` method produces a type implementing [`rayon`s `ParallelIterator`][ra]
+There is always overhead in parallelization, so you should always carefully profile to see if there are benefits in the switch. If you have only few things to iterate over then sequential join is faster.
```
Comment (torkleyy) [inline book/src/08_parallel_join.md]: Please remove the second "always". I'd probably put this into a little "box" by placing "\>" before the text.

File: book/src/08_parallel_join.md [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
 As mentioned in the chapter dedicated on how to [dispatch][c3] systems,
 Specs automatically parallelizes system execution when there are non-conflicting
-system data requirements (every system data may only be accessed
-mutably once xor immutably from multiple systems).
+system data requirements (Two `System`s conflict if their `SystemData` needs access
+to the same resource where at least one of them needs write access to it).
 
 [c3]: ./03_dispatcher.html
 
+## Basic parallization
```
Comment (itmuckel) [inline book/src/08_parallel_join.md]: Typo in parallelization.

Comment (WaDelma) [reply book/src/08_parallel_join.md]: I constantly make that typo, thanks for spotting!

**Pull request merged**